Add Miniflare/startWorker() + `node:test` tutorial (#19254)
cloudflare/cloudflare-docs: src/content/docs/workers/wrangler/api.mdx

Wrangler offers APIs to programmatically interact with your Cloudflare Workers.

- [`unstable_startWorker`](#unstable_startworker) - Start a server for running integration tests against your Worker.
- [`unstable_dev`](#unstable_dev) - Start a server for running either end-to-end (e2e) or integration tests against your Worker.
- [`getPlatformProxy`](#getplatformproxy) - Get proxies and values for emulating the Cloudflare Workers platform in a Node.js process.

## `unstable_startWorker`

> This API exposes the internals of Wrangler's dev server, and allows you to customise how it runs. For example, you could use `unstable_startWorker()` to run integration tests against your Worker (the below example is with `node:test`, but this should work in any testing framework):

Reviewer: ```suggestion
This API exposes the internals of Wrangler's dev server, and allows you to customise how it runs. For example, you could use `unstable_startWorker()` to run integration tests against your Worker. This example uses `node:test`, but should apply to any testing framework:
```

Wrangler offers APIs to programmatically interact with your Cloudflare Workers.

> - [`unstable_startWorker`](#unstable_startworker) - Start a server for running integration tests against your Worker.

Reviewer 2: workers/testing/integration-testing/ should also mention startWorker and link to the below section. Or possibly that section is a better place for this? Or duplicate it in a snippet...?

> - [`unstable_dev`](#unstable_dev) - Start a server for running either end-to-end (e2e) or integration tests against your Worker.

Reviewer 2: Can you explain when to use `unstable_dev` vs `unstable_startWorker`?
Should `unstable_dev` testing be in the testing overview matrix?

Author: Clarified—`unstable_startWorker` should be used in favour of `unstable_dev`

- [`getPlatformProxy`](#getplatformproxy) - Get proxies and values for emulating the Cloudflare Workers platform in a Node.js process.

## `unstable_startWorker`

This API exposes the internals of Wrangler's dev server, and allows you to customise how it runs. For example, you could use `unstable_startWorker()` to run integration tests against your Worker (the below example is with `node:test`, but this should work in any testing framework):

```js
import assert from "node:assert";
import test, { after, before, describe } from "node:test";

> import { unstable_startWorker } from "wrangler";

Reviewer 3: Will we make `startWorker` stable in v4? The unstable prefix will likely discourage people from using it 🤔

Author: We should be able to stabilise it outside of v4—it's not a breaking change. But yeah, we should do that soonish!
